### ISSUE TITLE
ObjectDetectionCoral stops working when restarting the docker container.

### ISSUE THREAD

**Issue by mikegleasonjr** [codeproject/CodeProject.AI-ObjectDetectionCoral#7]
**Area of Concern**
- [ ] Server
- [x] Behaviour of `ObjectDetectionCoral`
- [ ] Installer
- [ ] Runtime [e.g. Python3.7, .NET]
- [ ] Module packages [e.g. PyTorch)
- [ ] Something else

**Describe the bug**
1. I am running a docker container with the `codeproject/ai-server` image with 2 volumes (so that I can restart it without having to reinstall/reconfigure everything):
  - `/etc/codeproject/ai`
  - `/app/modules`
2. I installed the `Object Detection (Coral)` module, which worked fine with my TPU. I called the API and verified that everything was working.
3. I stopped the container
4. I started back the container
5. The module now has this error: `objectdetection_coral_adapter.py: An exception occurred initialising the module: libedgetpu.so.1: cannot open shared object file: No such file or dir`

**Expected behavior**
The module restarts properly in the docker container.

I know what the problem actually is: In `ObjectDetectionCoral/install.sh`, there is a line to copy the shared library in `/usr/lib/x86_64-linux-gnu/` upon module installation:

```
cp "${moduleDirPath}/edgetpu_runtime/${edgetpu_folder}/k8/libedgetpu.so.1.0" "/usr/lib/x86_64-linux-gnu/libedgetpu.so.1.0"
```

Obvisouly when restarting the container, this change is lost since I do not mount `/usr/lib/x86_64-linux-gnu/` as a volume to be persistent across "reboots".

I don't know if modules have a startup hook where these steps of copying the shared object and running of `ldconfig` could be done there?

**Screenshots**
N/A

**Your System (please complete the following information):**
 - CodeProject.AI Server version: codeproject/ai-server:2.6.2
 - Object Detection (Coral) version: 2.2.0

**Additional context**
See Expected behavior

**Comment by matthewDDennis** [comment]
Installing the latest version of the Server, 2.9.0, should resolve the issue.
Since 2.8.0 the Server will check to see if the container Id has changed and rerun the install scripts to ensure all the required files and libs are initialized.

**Comment by cpfarhood** [comment]
I'm still seeing this challenge as of 2.9.5

**Comment by ChrisMaunder** [comment]
@mikegleasonjr You're correct in pointing out where the issue is. The fix Matthew made was to re-run the install.sh script when the server detects that the container it's running under isn't the original container. This should then re-run the 
``` bash
cp "${moduleDirPath}/edgetpu_runtime/${edgetpu_folder}/k8/libedgetpu.so.1.0" "/usr/lib/x86_64-linux-gnu/libedgetpu.so.1.0"
```
command and ensure the library is in the correct place.

Just to clarify some things: you say you stopped and restarted the container. Does that mean you just stopped and restarted, or does it mean you deleted and recreated the container? I'm assuming you mean just "stop and start".

Could you try opening a terminal into the container and checking if `/usr/lib/x86_64-linux-gnu/libedgetpu.so.1.0` exists? If not could you please try going to the /app/modules/ObjectDetectionCoral folder and running
``` bash
bash ../../setup.sh
```
and then checking if `/usr/lib/x86_64-linux-gnu/libedgetpu.so.1.0` exists after the re-install?

**Comment by cpfarhood** [comment]
New container with module installed
root@aiserver-coral-2:/usr/lib/x86_64-linux-gnu# ls *tpu*
libedgetpu.so.1  libedgetpu.so.1.0

After killing the container
root@aiserver-coral-2:/usr/lib/x86_64-linux-gnu# ls *tpu*
ls: cannot access '*tpu*': No such file or directory

After running setup.sh
root@aiserver-coral-2:/usr/lib/x86_64-linux-gnu# ls *tpu*
libedgetpu.so.1  libedgetpu.so.1.0

So how to force this to happen when needed?

**Comment by cpfarhood** [comment]
> Installing the latest version of the Server, 2.9.0, should resolve the issue. Since 2.8.0 the Server will check to see if the container Id has changed and rerun the install scripts to ensure all the required files and libs are initialized.
> 
> In the case of the Docker image, you need to pull the appropriate codeproject/ai-server image that matches your CPU/GPU configuration. The latest image is 2.8.0, but a 2.9.0 should be available soon.

I'm using kubernetes not docker, is it possible the container id is not changing? I'm using a statefulset so the name is always the same on purpose.

**Comment by ChrisMaunder** [comment]
That could be it...

**Comment by cydia2020** [comment]
This can be solved by installing libedgetpu on the host, and passing the lib into the container by adding the following lines to the compose file
```yaml
    volumes:
      - /usr/lib/x86_64-linux-gnu/libedgetpu.so.1:/usr/lib/x86_64-linux-gnu/libedgetpu.so.1:ro
```
